gh-actions-workshop/presentation.md

```diff
@@ -0,0 +1,522 @@
+  - `cache` action is a generalized caching action that takes a list of paths and caches them with an explicit `key` for a cache entry. [Documentation](https://github.com/actions/cache).
+
+```yaml
+jobs:
+  job1:
+    steps:
+      - name: Cache Files
+        uses: actions/cache@v4
+        id: cache-files
+        with:
+          path:
+            - foo/
+            - bar
+          # caches by OS but also caches by current git ref
+          key: ${{ runner.os }}-${{ github.ref }}
+      - if: steps.cache-primes.outputs.cache-hit != 'true'
+        run: ./create-files.sh
+```
+
+---
+
+### GitHub Actions - Artifacts
+
+You sometimes wish to store files after a _job_/_workflow_ has been completed (e.g.: you have a workflow that builds an application but, you need the output to another workflow that deploys the new version).
+
+This is different from caching because caching is used when something is not changed often. However, artifacts are used when you want to save files like binaries or build logs. These are persisted, by default, for 90 days.
+
+```yaml
+jobs:
+  job1:
+    steps:
+      - name: Upload final apk
+        uses: actions/upload-artifact@v4
+        with:
+          name: final-apk-build
+          path: output/output.apk
+  job2:
+    needs: job1
+    steps:
+      - name: Download a .apk
+        uses: actions/download-artifact@v4
+        with:
+          name: final-apk-build
+```
+
+---
+
+### Examples from NIAEFEUP
+
+Sometimes the best way to learn is by looking at how the community implements CI/CD, these are some of our examples:
+
+- [UNI workflows](https://github.com/NIAEFEUP/uni/tree/develop/.github/workflows)
+- [New Website frontend workflows](https://github.com/NIAEFEUP/website-niaefeup-frontend/tree/develop/.github/workflows)
+- [New Website backend workflows](https://github.com/NIAEFEUP/website-niaefeup-backend/tree/develop/.github/workflows)
+
+---
+
+class: inverse, center, middle
+
+# Hands-on
+### I hope that wasn't too overwhelming 😅
+
+---
+
+### The objective
+
+Let's imagine you've recently entered a company that develops a Flutter application, CineScope (completely not my ESOF project) however they have no CI/CD processes and they would like you to implement it.
+
+You should have:
+ - A CI pipeline that runs the linter, the formatter, and finally, the unit tests. It should run on every PR but also every push to the main branch;
+ - A CD pipeline that builds the final application as an APK and creates a GitHub Release for it.
+
+---
+
+### How to
+
+You should create a new repository that is a template of this repo:
+
+<div style="display:flex; flex-direction: column; justify-content: space-between; height:15rem">
+  <div style="display:flex; justify-content: center">
+    <a href="https://github.com/NIAEFEUP/workshop-ci-24">https://github.com/NIAEFEUP/workshop-ci-24</a>
```

Comment:
Can "-template" be added to the name of the repo? People browsing our repos may be confused and think it is the workshop itself. Also, in the other repo we could have a link to this page.